crates/router/src/db/dispute.rs

```diff
@@ -131,54 +135,739 @@ impl DisputeInterface for Store {
+        connector_dispute_id: &str,
     ) -> CustomResult<Option<storage::Dispute>, errors::StorageError> {
-        // TODO: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        Ok(self
+            .disputes
+            .lock()
+            .await
+            .iter()
+            .find(|d| {
+                d.merchant_id == merchant_id
+                    && d.payment_id == payment_id
+                    && d.connector_dispute_id == connector_dispute_id
+            })
+            .cloned())
     }
 
     async fn find_dispute_by_merchant_id_dispute_id(
         &self,
-        _merchant_id: &str,
-        _dispute_id: &str,
+        merchant_id: &str,
+        dispute_id: &str,
     ) -> CustomResult<storage::Dispute, errors::StorageError> {
-        // TODO: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let locked_disputes = self.disputes.lock().await;
+
+        let found_dispute = locked_disputes
+            .iter()
+            .find(|d| d.merchant_id == merchant_id && d.dispute_id == dispute_id)
+            .cloned();
+
+        if let Some(dispute) = found_dispute {
+            Ok(dispute)
+        } else {
+            Err(errors::StorageError::ValueNotFound(format!("No dispute available for merchant_id = {merchant_id} and dispute_id = {dispute_id}")))?
+        }
     }
 
     async fn find_disputes_by_merchant_id(
         &self,
-        _merchant_id: &str,
-        _dispute_constraints: api_models::disputes::DisputeListConstraints,
+        merchant_id: &str,
+        dispute_constraints: api_models::disputes::DisputeListConstraints,
     ) -> CustomResult<Vec<storage::Dispute>, errors::StorageError> {
-        // TODO: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let locked_disputes = self.disputes.lock().await;
+
+        Ok(locked_disputes
+            .iter()
+            .filter(|d| {
+                d.merchant_id == merchant_id
+                    && dispute_constraints
+                        .dispute_status
+                        .as_ref()
+                        .map(|status| status == &DisputeStatus::foreign_from(d.dispute_status))
+                        .unwrap_or(true)
+                    && dispute_constraints
+                        .dispute_stage
+                        .as_ref()
+                        .map(|stage| stage == &DisputeStage::foreign_from(d.dispute_stage))
+                        .unwrap_or(true)
+                    && dispute_constraints
+                        .reason
+                        .as_ref()
+                        .map(|reason| {
+                            d.connector_reason
+                                .as_ref()
+                                .map(|connector_reason| connector_reason == reason)
+                                .unwrap_or(true)
+                        })
+                        .unwrap_or(true)
```

Comment:
You should be able to use `and_then()` instead of the outer `map()` and remove the inner `unwrap_or()`

```suggestion
                        .and_then(|reason| {
                            d.connector_reason
                                .as_ref()
                                .map(|connector_reason| connector_reason == reason)
                        })
                        .unwrap_or(true)
```